### PR TITLE
feat: add to watch later feature added

### PR DESCRIPTION
**Pull request by paarthbajaj**

### CODE REVIEW

File: src/components/VideoBlock.js [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
             <div className="video-meta-info flex-row">
-              <span className="video-views">100M views</span>
+              <span className="video-views">{views}</span>
               <span>•</span>
-              <span className="video-age">12 days ago</span>
+              <span className="video-age">{published_on}</span>
             </div>
           </div>
-          <i class="far fa-ellipsis-v action-menu"></i>
+          {/* <i className="far fa-ellipsis-v action-menu"></i> */}
```
Comment (neetukumari4858) [inline src/components/VideoBlock.js]: remove this commented line

**Review by neetukumari4858 — APPROVED** [review]
code is well Structured and readable
overall looks fine to me 
one suggession is that Add description and remove commented line
good job @paarthbajaj

File: src/components/AddToPlaylistPopup.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          </div>
+          {videoState?.listOfPlaylist.map((i) => (
+            <div className="list-playlist" key={i}>
+              <label>
+                <input type="checkbox" />
+                {i}
+              </label>
+            </div>
+          ))}
+          {/* {videoState.showCreatePlaylistBlock && ( */}
```
Comment (ankit9015) [inline src/components/AddToPlaylistPopup.js:29]: Remove commented code before merge

File: src/components/AddToPlaylistPopup.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                videoDispatch({ type: "HIDE_CREATE_PLAYLIST_BLOCK" });
+                videoDispatch({ type: "CLOSE_PLAYLIST_POPUP" });
+                addToPlaylist(playlistName);
+              }}
+            >
+              Create
+            </button>
+          </div>
+          {/* )} */}
+          {/* {!videoState.showCreatePlaylistBlock && ( */}
```
Comment (ankit9015) [inline src/components/AddToPlaylistPopup.js:52]: Here also

File: src/components/AddToPlaylistPopup.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+          <div
+            className="popup-footer cursor-pointer"
+            onClick={() =>
+              videoDispatch({ type: "SHOW_CREATE_PLAYLIST_BLOCK" })
+            }
+          >
+            <i className="far fa-plus"></i>
+            <span>Create a new playlist</span>
+          </div>
+          {/* )} */}
```
Comment (ankit9015) [inline src/components/AddToPlaylistPopup.js:62]: here

File: src/components/WatchLater.js [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
+  }, []);
   return (
     <>
       <div className="app-body">
         <Sidebar />
         <div className="vl-page-container">
           <h1 className="txt-3">Watch Later</h1>
           <div className="video-list-container mt-1 flex-row g-1">
-            <VideoBlock />
+            {videoState?.watchLaterList?.map((video) => (
```
Comment (ankit9015) [inline src/components/WatchLater.js:18]: I like that you are using optional chaining

**Review by ankit9015 — COMMENTED** [review]
Overall code is good to merge

**Review by ankit9015 — APPROVED** [review]

File: src/components/VideoBlock.js (hunk 1 above)
Comment (paarthbajaj) [reply src/components/VideoBlock.js]: Done

File: src/components/AddToPlaylistPopup.js (hunk 2 above)
Comment (paarthbajaj) [reply src/components/AddToPlaylistPopup.js:29]: The logic will be implemented in future, that's why haven't removed the commented code.

**Pull request merged**